spec/custom_attribute_condition_evaluator_spec.rb

```diff
@@ -0,0 +1,271 @@
+    expect(condition_evaluator.evaluate('match' => 'invalid', 'name' => 'weird_condition', 'type' => 'custom_attribute', 'value' => 'bye')).to eq(nil)
+  end
+
+  describe 'exists match type' do
+    before(:context) do
+      @exists_conditions = {'match' => 'exists', 'name' => 'input_value', 'type' => 'custom_attribute'}
+    end
+
+    it 'should return false if there is no user-provided value' do
+      condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new({})
+      expect(condition_evaluator.evaluate(@exists_conditions)).to be false
+    end
+
+    it 'should return false if the user-provided value is nil' do
+      condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('input_value' => nil)
+      expect(condition_evaluator.evaluate(@exists_conditions)).to be false
+    end
+
+    it 'should return true if the user-provided value is a string' do
+      condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('input_value' => 'test')
+      expect(condition_evaluator.evaluate(@exists_conditions)).to be true
+    end
+
+    it 'should return true if the user-provided value is a number' do
+      condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('input_value' => 10)
+      expect(condition_evaluator.evaluate(@exists_conditions)).to be true
+    end
+
+    it 'should return true if the user-provided value is a boolean' do
+      condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('input_value' => false)
+      expect(condition_evaluator.evaluate(@exists_conditions)).to be true
+    end
+  end
+
+  describe 'exact match type' do
+    describe 'with a string condition value' do
+      before(:context) do
+        @exact_string_conditions = {'match' => 'exact', 'name' => 'location', 'type' => 'custom_attribute', 'value' => 'san francisco'}
+      end
+
+      it 'should return true if the user-provided value is equal to the condition value' do
+        condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('location' => 'san francisco')
+        expect(condition_evaluator.evaluate(@exact_string_conditions)).to be true
+      end
+
+      it 'should return false if the user-provided value is not equal to the condition value' do
+        condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('location' => 'new york')
+        expect(condition_evaluator.evaluate(@exact_string_conditions)).to be false
+      end
+
+      it 'should return nil if the user-provided value is of a different type than the condition value' do
+        condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('location' => false)
+        expect(condition_evaluator.evaluate(@exact_string_conditions)).to eq(nil)
+      end
+
+      it 'should return nil if there is no user-provided value' do
+        condition_evaluator = Optimizely::CustomAttributeConditionEvaluator.new('location' => {})
```

Comment:
This checks the same as in the above test case. 
Send empty hash as user attributes to verify